File: server/apps/api/event/v1/serializers/event.py
from rest_framework import serializers
from apps.api.event.models import Event

class EventBasicSerializer(serializers.ModelSerializer):

    class Meta:
        model = Event
        fields = ["id", "name", "description", "image"]


class EventDetailSerializer(serializers.ModelSerializer):

    author = serializers.CharField(source="author.username")
    class Meta:
        model = Event
        fields = "__all__"

File: src/models/event.ts
export interface Event {
  id: number | undefined
  name: string | undefined
  description: string | undefined
  image: string | undefined
}
export interface EventDetail {
  id: number | undefined
  event: Event
  date: string | undefined
  address: string | undefined
  landmark: string | undefined
  author: string | undefined
  description?: string | undefined
}
